### ISSUE TITLE
/v1/update API endpoint should apply pending target state steps

### ISSUE THREAD

**Issue by pipex** [balena-io/balena-supervisor#1530]
Currently if an update is pending, there is no way to trigger the update by the user after releasing the lock. This is what it is expected from [the documentation](https://www.balena.io/docs/reference/supervisor/supervisor-api/#post-v1update)

Steps to reproduce
* Set an update lock on a service
* Push a new release
 * Wait untill the supervisor reports Updates are locked, retrying in 15m...
 * Remove the lock and call curl -X POST --header "Content-Type:application/json" "$BALENA_SUPERVISOR_ADDRESS/v1/update?apikey=$BALENA_SUPERVISOR_API_KEY"
 * The endpoint replies with 204 but the update is not applied (even using force:true)

**Comment by 20k-ultra** [comment]
What's happening in the **/v1/update** endpoint is it checks if the target state we have on device is the same as the cloud. If the cloud API says the target states are they same then the logic ends and returns 204. However, if the target state on the device and the cloud do not match then this target state is stored into memory and applied. 

This separation of checking then applying is important because the timeout interval only exists in the applying portion. Therefore, when using this endpoint it doesn't apply the target state because it already is! This is why you get the messaging saying that locks were found so retrying in X time.

To me the endpoint is working correctly as it has done its job of updating the device target state. This issue should be about trying to improve the UX though. When a **/v1/update** endpoint is received it should perform the target state checks (does local version match cloud) but also try to reset the interval for applying as well.

Regarding `force: true` not working btw, it does work. I created a lock on my service then disconnected my device from the internet and changed the target release. After I connected it to the internet again and sent **/v1/update** with `force: true` (note I did not remove the lock file) the device changed to the new target. The issue is if you send this request but a timeout interval is already running. The force isn't to short circuit this timer, it's just to ignore the lock files. 

In short the goal is to make the **/v1/update** endpoint clear the existing timer so that the UX is improved.

**Comment by 20k-ultra** [comment]
Will be submitting a PR for this soon just need to add some unit testing but I got it working now where the /v1/update will short circuit the existing timeout and try to apply the target state right away. If there is still a lock then the timer just starts from the beginning (1 second).
